### PR TITLE
Replace loop with LastIndexOfAny in DeveloperExperience.cs

### PR DESCRIPTION
**Pull request by MihaZupan**

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs [hunk 1]
```diff
@@ -134,14 +134,11 @@ private static string GetFileNameWithoutExtension(string path)
 
         private static string GetFileName(string path)
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs:135]: I would be better to delete GetFileName and GetFileNameWithoutExtension, and replace it with a call to `Path.GetFileNameWithoutExtension`

Comment (stephentoub) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs:135]: I was going to ask about that, but then figured there was a reason they were being duplicated here.  I guess not?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs [hunk 2]
```diff
@@ -65,7 +66,7 @@ public virtual string CreateStackTraceString(IntPtr ip, bool includeFileInfo)
             }
 
             StringBuilder sb = new StringBuilder();
-            string fileNameWithoutExtension = GetFileNameWithoutExtension(moduleFullFileName);
+            string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(moduleFullFileName);
             int rva = (int)(ip.ToInt64() - moduleBase.ToInt64());
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs:70]:
```suggestion
            ReadOnlySpan<char> fileNameWithoutExtension = Path.GetFileNameWithoutExtension(moduleFullFileName.AsSpan());
            int rva = (int)(ip - moduleBase);
```
Avoid allocation and simplify

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**